RecoJets/JetProducers/plugins/HTTTopJetProducer.cc

```diff
@@ -0,0 +1,244 @@
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "DataFormats/JetReco/interface/HTTTopJetTagInfo.h"
+#include "DataFormats/JetReco/interface/BasicJetCollection.h"
+#include "HTTTopJetProducer.h"
+
+using namespace edm;
+using namespace cms;
+using namespace reco;
+using namespace std;
+
+HTTTopJetProducer::HTTTopJetProducer(edm::ParameterSet const& conf):
+       FastjetJetProducer( conf ),
+       optimalR_(false),
+       qJets_(false),
+       minFatjetPt_(200.),
+       minSubjetPt_(20.),
+       minCandPt_(200.),
+       maxFatjetAbsEta_(2.5),
+       subjetMass_(30.),
+       muCut_(0.8),
+       filtR_(0.3),
+       filtN_(5),       
+       mode_(0),
+       minCandMass_(150.),
+       maxCandMass_(200.),
+       massRatioWidth_(15),
+       minM23Cut_(0.35),
+       minM13Cut_(0.2),
+       maxM13Cut_(1.3),
+       maxR_(1.5),
+       minR_(0.5),
+       rejectMinR_(false),
+       verbose_(false )
+{
+  
+  // Read in all the options from the configuration
+  if ( conf.exists("optimalR") ) 
```

Comment:
looks like this rule is generally violated in the base class of this producer.
So, the requirement to use  fillDescriptions in this derived class is less strong.